File: laravel-api/app/Helpers/ApiResponse.php
<?php

namespace App\Helpers;

use Illuminate\Http\JsonResponse;

class ApiResponse
{
    public static function success($data = null, string $message = 'OK', int $code = 200): JsonResponse
    {
        return response()->json([
            'success' => true,
            'message' => $message,
            'data'    => $data,
        ], $code);
    }
}

File: flutter_app/lib/core/services/response/api_response.dart
class ApiResponse<T> {
  final bool success;
  final String message;
  final T? data;
  final List<String>? errors;

  ApiResponse({
    required this.success,
    required this.message,
    this.data,
    this.errors,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json, {
    T Function(Map<String, dynamic>)? fromJsonT,
  }) {
    List<String>? extractedErrors;
    if (json['errors'] != null && json['errors'] is Map) {
      extractedErrors = (json['errors'] as Map<String, dynamic>)
          .values
          .expand((value) => List<String>.from(value))
          .toList();
    }
    T? parsedData;
    if (fromJsonT != null && json['data'] != null) {
      parsedData = fromJsonT(Map<String, dynamic>.from(json['data']));
    }
    return ApiResponse<T>(
      success: json['success'] ?? false,
      message: json['message'] ?? '',
      data: parsedData,
      errors: extractedErrors,
    );
  }
}
